corehq/apps/users/models.py

```diff
@@ -2029,30 +2020,11 @@ def get_role(self, domain=None, include_teams=True, checking_global_admin=True):
         if not include_teams:
             return super(WebUser, self).get_role(domain)
 
-        dm_list = list()
-
         dm = self.get_domain_membership(domain)
-        if dm:
-            dm_list.append([dm, ''])
-
-        # now find out which dm has the highest permissions
-        if dm_list:
-            return self.total_domain_membership(dm_list, domain)
-        else:
+        if not dm:
             raise DomainMembershipError()
 
-    def total_domain_membership(self, domain_memberships, domain):
-        #sort out the permissions
-        total_permission = Permissions()
-        if domain_memberships:
-            for domain_membership, membership_source in domain_memberships:
-                permission = domain_membership.permissions
-                total_permission |= permission
```

Comment:
This bit is the important part.  I assume at some point some users must've had multiple domain memberships, but now this function is only called in two places, both of which pass in a list of length 1 for `domain_memberships` (and both are modified in this PR).

What remained was this weird behavior, where we'd take their assigned permissions and compare that against the default `Permissions()` instance, and creating a new, fake user role with the more permissive of each property.  This works fine now, since the default permissions all disallow everything, but it's very strange that users don't actually get the role they're assigned, exactly.

Additionally, if we ever add any permissions that default to the *more* permissive option (as is the case in my unrelated change), this would always give the users that permission, even when their role expressly disallows it.